addons/stock/report/report_stock_rule.py

```diff
@@ -0,0 +1,96 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, models, _
+from odoo.exceptions import UserError
+
+
+class ReportStockRule(models.AbstractModel):
+    _name = 'report.stock.report_stock_rule'
+
+    @api.model
+    def get_report_values(self, docids, data=None):
+        product = self.env['product.product'].browse(data['product_id'])
+        warehouses = self.env['stock.warehouse'].browse(data['warehouse_ids'])
+
+        routes = self._get_routes(data)
+
+        rules = routes.mapped('rule_ids').filtered(lambda r: r.warehouse_id == False or r.warehouse_id in (warehouses))
```

Comment:
add a comment:
some routes do not have a warehouse_id but contains rules for different warehouses ; make sure we only display the rules of the selected warehouses

also you don't need the parens around the last `warehouses`